specification/modsimworkbench/resource-manager/Microsoft.ModSimWorkbench/preview/2021-03-01-preview/modsimworkbenches.json

```diff
@@ -0,0 +1,636 @@
+        "Unknown",
+        "Validating",
+        "Processing",
+        "Deleting",
+        "Accepted",
+        "Canceled",
+        "Completed",
+        "Succeeded",
+        "Failed",
+        "Deleted"
+      ],
+      "type": "string",
+      "readOnly": true
+    },
+    "ErrorResponse": {
+      "type": "object",
+      "title": "Error response",
+      "description": "Common error response for all Azure Resource Manager APIs to return error details for failed operations. (This also follows the OData error response format.).",
+      "properties": {
+        "error": {
+          "description": "The error object.",
+          "$ref": "#/definitions/ErrorDetail"
+        }
+      }
+    },
+    "ErrorAdditionalInfo": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The additional info type."
+        },
+        "info": {
+          "readOnly": true,
+          "type": "object",
+          "description": "The additional info."
+        }
+      },
+      "description": "The resource management error additional info."
+    },
+    "Resource": {
+      "type": "object",
+      "description": "Common resource representation.",
+      "x-ms-azure-resource": true,
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource ID."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource name."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type."
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags."
+        },
+        "systemData": {
+          "description": "system data",
+          "title": "system data",
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/systemData"
+        },
+        "sku": {
+          "type": "string",
+          "description": "Name of SKU."
+        }
+      }
+    },
+    "CidrBlock": {
+      "description": "Specifies the IP address range for ModSimWorkbench workbench will use in CIDR notation. Avoid overlapping or duplicated IP address range with other ModSimWorkbench workbenches and on-premise networks when connecting to on-premise networks using VPN or ExpressRoute.",
+      "type": "string"
+    },
+    "AppOidc": {
+      "description": "Workbench properties.",
+      "type": "object",
+      "properties": {
+        "clientId": {
+          "type": "string",
+          "description": "Specifies the key vault secret identifier for Azure AD app client secret ID."
+        },
+        "secretUri": {
```

Comment:
>secretUri [](http://example.com/codeflow?start=9&length=9)

Does user need to set up the permissions so that your RP will be able to read the secret from their vault? How would they do it in a ARM API / deployment template scenario?